ds3-contract-comparator-cli/src/main/java/com/spectralogic/ds3contractcomparator/cli/CLI.java

```diff
@@ -66,14 +70,26 @@ private Arguments processArgs(final String[] args) throws ParseException {
         final boolean help = cmd.hasOption("h");
         final boolean properties = cmd.hasOption("properties");
         final boolean annotations = cmd.hasOption("annotations");
+        final PrinterType printerType = processPrinterType(cmd);
 
-        final Arguments arguments = new Arguments(oldSpec, newSpec, outFile, help, properties, annotations);
+        final Arguments arguments = new Arguments(oldSpec, newSpec, outFile, help, properties, annotations, printerType);
 
         validateArguments(arguments);
 
         return arguments;
     }
 
+    private PrinterType processPrinterType(final CommandLine cmd) {
+        try {
+            return Guards.returnIfNull(cmd.getOptionValue("p").toUpperCase(), PrinterType::valueOf);
+        } catch (final NullPointerException e) {
```

Comment:
When would there be a NPE?  I typically would not use NPEs for error validation.